src/event.rs

```diff
@@ -9,8 +9,59 @@
 use crate::routing_table::Authority;
 use crate::routing_table::Prefix;
 use crate::xor_name::XorName;
+use crate::NetworkBytes;
 use hex_fmt::HexFmt;
+use quic_p2p::Token;
 use std::fmt::{self, Debug, Formatter};
+use std::net::SocketAddr;
+
+/// An Event raised by a `Client`
+///
+/// These are send transparently to the user library and not handled by routing
+#[derive(Clone, Eq, PartialEq)]
+pub enum ClientEvent {
+    /// Inform the user (library) that we are connected to a client
+    ConnectedToClient {
+        /// Client's endpoint
+        peer_addr: SocketAddr,
+    },
+    /// Inform the user (library) that we are disconnected from a client
+    ConnectionFailureToClient {
+        /// Client's endpoint
+        peer_addr: SocketAddr,
+    },
+    /// Inform the user (library) that we have a new message from a client
+    NewMessageFromClient {
+        /// Client's endpoint
+        peer_addr: SocketAddr,
+        /// Client's message
+        msg: NetworkBytes,
+    },
+    /// Inform the user (library) that we couldn't send this message to a client
+    UnsentUserMsgToClient {
+        /// Client's endpoint
+        peer_addr: SocketAddr,
+        /// Message we had tried to send to the client
+        msg: NetworkBytes,
+        /// Token that we had used to identify this message
+        token: Token,
+    },
+    /// Inform the user (library) that we have successfully sent this message to a client
+    SentUserMsgToClient {
+        /// Client's endpoint
+        peer_addr: SocketAddr,
+        /// Message we had tried to send to the client
+        msg: NetworkBytes,
+        /// Token that we had used to identify this message
+        token: Token,
+    },
+}
+
+impl Into<Event> for ClientEvent {
+    fn into(self) -> Event {
+        Event::ClientEvent(self)
+    }
+}
```

Comment:
Nitpick, but it's more idiomatic to define this as `impl From<ClientEvent> for Event`. Then you get the `Into` impl for free.